dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -54,641 +46,237 @@
-                            (uri, siteId, languageId);
+  }
 
-            if (null == liveCachedVanityUrl) {
 
-                liveCachedVanityUrl =
-                        this.getFallback(uri, siteId, languageId, this.systemUser);
-            }
-
-            isVanityURL =
-                    !is404(liveCachedVanityUrl);
-
-            // Still support legacy cmsHomePage
-            if (URL_SUFFIX.equals(uri) && !isVanityURL) {
-
-                liveCachedVanityUrl =
-                        this.getLiveCachedVanityUrl
-                                (LEGACY_CMS_HOME_PAGE, host, languageId, this.systemUser);
-                isVanityURL = UtilMethods.isSet(liveCachedVanityUrl)
-                        && !VanityUrlAPI.CACHE_404_VANITY_URL
-                            .equals(liveCachedVanityUrl.getVanityUrlId());
-            }
-        }
-
-        return isVanityURL;
-    } // isVanityUrl
-
-    @CloseDBIfOpened
-    @Override
-    public VanityUrl getVanityUrlFromContentlet(final Contentlet contentlet) {
-
-        if (contentlet != null) {
-            try {
-                if (!contentlet.isVanityUrl()) {
-                    throw new DotStateException(
-                            "Contentlet : " + contentlet.getInode() + " is not a Vanity Url");
-                }
-            } catch (DotDataException | DotSecurityException e) {
-                throw new DotStateException(
-                        "Contentlet : " + contentlet.getInode() + " is not a Vanity Url", e);
-            }
-        } else {
-            throw new DotStateException("Contentlet is null");
-        }
-
-        final DefaultVanityUrl vanityUrl = new DefaultVanityUrl();
-        vanityUrl.setContentTypeId(contentlet.getContentTypeId());
-        try {
-            this.contentletAPI.copyProperties(vanityUrl, contentlet.getMap());
-        } catch (Exception e) {
-            throw new DotStateException("Vanity Url Copy Failed", e);
-        }
-        vanityUrl.setHost(contentlet.getHost());
-        vanityUrl.setFolder(contentlet.getFolder());
-
-        return vanityUrl;
-    } // getVanityUrlFromContentlet.
-
-    /**
-     * Add the Vanity URL to the vanityURLCache, without affecting any secondary cache.
-     *
-     * @param vanityUrl The vanityurl URL object
-     */
-    private void addToSingleVanityURLCache(final VanityUrl vanityUrl) {
-        try {
-            if (vanityUrl.isLive()) {
-                vanityUrlServices.addSingleCache(vanityUrl);
-            } else {
-                vanityUrlServices.invalidateVanityUrl(vanityUrl);
-            }
-        } catch (DotDataException | DotRuntimeException | DotSecurityException e) {
-            Logger.error(this,
-                    "Error trying to add Vanity URL identifier:" + vanityUrl.getIdentifier()
-                            + " to VanityURLCache", e);
+  
+  
+  
+  private List<CachedVanityUrl> load(Host host, Language lang){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'lang' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)